### PR TITLE
FEATURE: Add menu to assigned text on posts

### PR DESCRIPTION
**Pull request by AndrewPrigorshnev**
This adds a popup menu to assigned posts that makes it easier to unassign or reassign them:

Screenshot 2024-03-08 at 00 22 23

It was pretty difficult to write the tests. I've been trying both acceptance and system specs and decided to stick with acceptance tests, though one acceptance spec is still not passing, and I skipped it for now. We'll be adding more similar menus in following PRs, and I'm going to invest more time into tests stability while working on that.

I haven't included [the system specs](https://github.com/AndrewPrigorshnev/discourse-assign/commit/977faa9bc74871d347a198d7067827c3f4d838d3) I wrote into this PR for now (they don't pass) but we may decide to use them, let's see how adding other menus in following PRs will go.

### CODE REVIEW

**Review by chapoi — APPROVED** [review]

File: assets/javascripts/discourse/components/assigned-to-post.gjs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    <a href={{@href}} class="assigned-to-username">
+      {{#if @assignedToUser}}
+        {{@assignedToUser.username}}
+      {{else}}
+        {{@assignedToGroup.name}}
+      {{/if}}
+    </a>
+
+    <DMenu @icon="ellipsis-h" class="btn-flat more-button">
+      <div class="popup-menu">
```
Comment (jjaffeux) [inline assets/javascripts/discourse/components/assigned-to-post.gjs:42]: is this necessary ?

Comment (AndrewPrigorshnev) [reply assets/javascripts/discourse/components/assigned-to-post.gjs:42]: Yes, we need that to style the popup menu appropriately.

Comment (AndrewPrigorshnev) [reply assets/javascripts/discourse/components/assigned-to-post.gjs:42]: (`DMenu` doesn't provide that)

**Comment by AndrewPrigorshnev** [comment]
@jjaffeux, I'm ready to merge it as soon as you approve.

**Review by jjaffeux — APPROVED** [review]

**Comment by jjaffeux** [comment]
> @jjaffeux, I'm ready to merge it as soon as you approve.

approved!

**Pull request merged**